contracts/tests/test_unruggable_memecoin.cairo

```diff
@@ -45,3 +45,324 @@ fn test_mint() {
+
+
+    // index starts from 1 because the owner
+    // is considered to be the first hodler
+    let mut index = 1; 
+    loop {
+        if index == MAX_HOLDERS_BEFORE_LAUNCH {
+            break;
+        }
+
+        // make transfer 
+        let recipient: ContractAddress = (index.into() + 9999999).try_into().unwrap();
+        safe_dispatcher.transfer(recipient, 1);
+
+        // check initial balance. Should be equal to initial supply.
+        let recipient_balance = safe_dispatcher.balance_of(recipient);
+        assert(recipient_balance == 1, 'invalid recipient balance');
+
+        index += 1;
+    };
+}
+
+
+#[test]
+#[should_panic(expected: ('Unruggable: max holders reached', ))]
+fn test_transfer_3() {
+    /// Ensure that transfers can only be made out to 
+    /// up to `MAX_HOLDERS_BEFORE_LAUNCH` addresses
+
+    let owner = contract_address_const::<42>();
+    let initial_supply = 1000.into();
+    let contract_address = deploy_contract(
+        owner, owner, 'UnruggableMemecoin', 'MT', initial_supply
+    );
+
+    // set owner as caller
+    start_prank(CheatTarget::One(contract_address), owner);
+
+    let safe_dispatcher = IUnruggableMemecoinDispatcher { contract_address };
+
+    let mut index = 0; 
+    loop {
+        if index == MAX_HOLDERS_BEFORE_LAUNCH + 1 {
+            break;
+        }
+
+        // make transfer 
+        let recipient: ContractAddress = (index.into() + 9999999).try_into().unwrap();
+        safe_dispatcher.transfer(recipient, 1);
+
+        // check initial balance. Should be equal to initial supply.
+        let recipient_balance = safe_dispatcher.balance_of(recipient);
+        assert(recipient_balance == 1, 'invalid recipient balance');
+
+        index += 1;
+    };
+}
+
+
+
+#[test]
+fn test_transfer_4() {
+    /// Ensure that transfer to more than `MAX_HOLDERS_BEFORE_LAUNCH`
+    /// works after the token is launched
+
+    let owner = contract_address_const::<42>();
+    let initial_supply = 1000.into();
+    let contract_address = deploy_contract(
+        owner, owner, 'UnruggableMemecoin', 'MT', initial_supply
+    );
+
+
+    // set owner as caller
+    start_prank(CheatTarget::One(contract_address), owner);
+
+    let safe_dispatcher = IUnruggableMemecoinDispatcher { contract_address };
+    // launch memecoin
+    safe_dispatcher.launch_memecoin();
+
+    let mut index = 0; 
+    loop {
+        if index == MAX_HOLDERS_BEFORE_LAUNCH + 1 {
+            break;
+        }
+
+        // make transfer 
+        let recipient: ContractAddress = (index.into() + 9999999).try_into().unwrap();
+        safe_dispatcher.transfer(recipient, 1);
+
+        // check initial balance. Should be equal to initial supply.
+        let recipient_balance = safe_dispatcher.balance_of(recipient);
+        assert(recipient_balance == 1, 'invalid recipient balance');
+
+        index += 1;
+    };
+}
+
+
+
+#[test]
+fn test_transfer_from() {
```

Comment:
all the transfer_from functions can be removed in favor of the testing of the new `_transfer` function that performs the tests directly